File: src/apnacollegeDSA/Sorting/bubbleSort.java
package apnacollegeDSA.Sorting;

public class bubbleSort {
    public static void bubbleSorting(int[] arr) {
        for (int turns = 0; turns <arr.length; turns++) {
            for (int i = 0; i < arr.length - 1 - turns ; i++) {
                if (arr[i] > arr[i+1]) {
                    //swapping
                    int temp = arr[i];
                    arr[i] = arr[i+1];
                    arr[i+1] = temp;
                }
            }
        }
    }
    public static void printArr(int arr[]){
        for (int i = 0; i < arr.length ; i++) {
            System.out.print(arr[i] +" ");
        }
    }
    public static void main(String[] args) {
        int arr[] = {3, 35, 10, 25, 15};
        bubbleSorting(arr);
        printArr(arr);
    }
}

File: C Practical/c tut/bubblesort.c
#include <stdio.h>

void printarr(int arr[], int size) {
    for (int i = 0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

void bubblesort(int arr[], int size) {
    for (int i = 0; i < size - 1; i++) {
        for (int j = 0; j < size - 1 - i; j++) { // Corrected 'i' to 'j'
            int temp;
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    int arr[100] = {20, 10, 60, 80, 50, 60};
    int n = 6;

    printarr(arr, n);
    bubblesort(arr, n);
    printarr(arr, n);
    
    return 0;
}